internal/providers/terraform/aws/eks_node_group.go

```diff
@@ -0,0 +1,138 @@
+		},
+	}
+}
+
+func NewEKSNodeGroup(d *schema.ResourceData, u *schema.ResourceData) *schema.Resource {
+	region := d.Get("region").String()
+	scalingConfig := d.Get("scaling_config").Array()[0]
+	desiredSize := scalingConfig.Get("desired_size").Int()
+	instanceType := "t3.medium"
+	if d.Get("instance_types").Exists() {
+		// Only a single type is expected https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_node_group#instance_types
+		instanceType = d.Get("instance_types").Array()[0].String()
+	}
+
+	launchTemplateRef := d.References("launch_template.0.id")
+
+	costComponents := make([]*schema.CostComponent, 0)
+	subResources := make([]*schema.Resource, 0)
+
+	if len(launchTemplateRef) > 0 {
+		onDemandCount := decimal.NewFromInt(desiredSize)
+		spotCount := decimal.Zero
+		if launchTemplateRef[0].Get("instance_market_options.0.market_type").String() == "spot" {
+			onDemandCount = decimal.Zero
+			spotCount = decimal.NewFromInt(desiredSize)
+		}
+		lt := newLaunchTemplate(launchTemplateRef[0].Address, launchTemplateRef[0], region, onDemandCount, spotCount)
+
+		// AutoscalingGroup should show as not supported LaunchTemplate is not supported
+		if lt == nil {
+			return nil
+		}
+		subResources = append(subResources, lt)
+	} else {
+		costComponents = append(costComponents, eksComputeCostComponent(d, region, desiredSize, instanceType))
+		eksCPUCreditsCostComponent := eksCPUCreditsCostComponent(d, region, desiredSize, instanceType)
+		if eksCPUCreditsCostComponent != nil {
+			costComponents = append(costComponents, eksCPUCreditsCostComponent)
+		}
+		costComponents = append(costComponents, newEksRootBlockDevice(d))
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+		SubResources:   subResources,
+	}
+}
+
+func eksComputeCostComponent(d *schema.ResourceData, region string, desiredSize int64, instanceType string) *schema.CostComponent {
+
+	purchaseOptionLabel := "on-demand"
+
+	return &schema.CostComponent{
+		Name:           fmt.Sprintf("Linux/UNIX usage (%s, %s)", purchaseOptionLabel, instanceType),
+		Unit:           "hours",
+		UnitMultiplier: 1,
+		HourlyQuantity: decimalPtr(decimal.NewFromInt(desiredSize)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("aws"),
+			Region:        strPtr(region),
+			Service:       strPtr("AmazonEC2"),
+			ProductFamily: strPtr("Compute Instance"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "instanceType", Value: strPtr(instanceType)},
+				{Key: "operatingSystem", Value: strPtr("Linux")},
+				{Key: "preInstalledSw", Value: strPtr("NA")},
+				{Key: "tenancy", Value: strPtr("Shared")},
+				{Key: "capacitystatus", Value: strPtr("Used")},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("on_demand"),
+		},
+	}
+}
+
+func eksCPUCreditsCostComponent(d *schema.ResourceData, region string, desiredSize int64, instanceType string) *schema.CostComponent {
```

Comment:
Same comment as above, I think `instance.cpuCreditsCostComponent` could be reused?

Reply:
See https://github.com/infracost/infracost/pull/206#discussion_r520451694